README.md

```diff
@@ -66,74 +66,79 @@ You can add the content to your Ballerina service as shown below. In that code s
 
 ```ballerina
 import ballerina/http;
+import ballerina/log;
 
-endpoint http:Listener filterServiceEP {
-    port: 9090
-};
+listener http:Listener filterServiceEP = new(9090);
 
-endpoint http:Client stdInfoEP {
-    url: "http://www.mocky.io"
-};
+http:Client stdInfoEP = new("http://www.mocky.io");
 
 // REST service to select the passed student from an exam
-service<http:Service> filterService bind filterServiceEP {
+service filterService on filterServiceEP {
 
-    // Resource that handles HTTP POST request with JSON payload
+    // Resource that handle HTTP POST request with JSON payload
     // Response with JSON payload
     @http:ResourceConfig {
         methods: ["POST"],
         consumes: ["application/json"],
         produces: ["application/json"]
     }
-
-    filterMarks (endpoint caller, http:Request request) {
-        http:Response response;
-
-        // Get the JSON payload from the request
-        json reqPayload = check request.getJsonPayload();
-
-        // Get the information of the subjects
-        string stdName = check <string>reqPayload.name;
-        json subjects = reqPayload.subjects;
+    resource function filterMarks (http:Caller caller, http:Request request) {
 
         // Declare boolean flag to set Qualified or Not
         boolean isQualified = false;
 
-        // Iterating subject array
-        foreach subj in subjects {
-            int mark = check <int>subj.marks;
-            // Check if the student exceeded the pass mark value
-            if (mark >= 60) {
-                isQualified = true;
-            }else{
-                isQualified = false;
-            }
-        }
-        // Set original payload to a new request object
+        // Set Original payload to a new request object
         http:Request req = new;
-        req.setJsonPayload(untaint reqPayload);
 
-        // Define variables for response payload and status code
+        // Get the JSON payload from the request
+        var reqPayload = request.getJsonPayload();
+        if (reqPayload is json) {
+            // Get the information of the subjects
+            string stdName = <string>reqPayload.name;
+            json[] subjects = <json[]>reqPayload.subjects;
+
+            // Iterating subject array
+            foreach var subj in subjects {
+                int mark = <int>subj.marks;
+                // Check the student exceed the pass mark value
+                if (mark >= 60) {
+                    isQualified = true;
+                } else {
+                    isQualified = false;
+                }
+            }
+            req.setJsonPayload(untaint reqPayload);
+        } else if (reqPayload is error) {
+            log:printError("Invalid request payload: Not a json", err = reqPayload);
```

Comment:
This `Not a json` doesn't seem to be correct here. WDYT? This can be a json but the parser can fail. Just because the json is invalid it doesn't seem right to say its not a json. Better to rephrase this to say either its an invalid payload or parser error occurs or something along the line